### PR TITLE
Border styles [MAP-661]

### PR DESCRIPTION
**Pull request by janbaykara**
Inherits from #148 

- Make political borders crisper and clearer against the default map
- Add backdrop to the area label names
- Use the non-draft style in production

# After
Screenshot 2024-12-09 at 15 49 05

# Before
Screenshot 2024-12-09 at 15 54 58

### CODE REVIEW

File: nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts [hunk 1]
```diff
@@ -23,7 +27,7 @@ const useBoundaryAnalytics = (
   let variables: any = {
     reportID: report?.id,
   }
```
Comment (ev-sc) [inline nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts:29]: What if this isn't a constituency?

File: nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts [hunk 2]
```diff
@@ -37,12 +41,16 @@ const useBoundaryAnalytics = (
     dataOutputKey = 'importedDataCountByConstituency'
   } else if (boundaryType === 'admin_ward') {
     query = MAP_REPORT_WARD_STATS
+    // @ts-ignore — asserting here that importedDataCountByWard will also return GroupedDataCount[]
     dataOutputKey = 'importedDataCountByWard'
   } else throw new Error('Invalid boundary type')
 
-  const boundaryAnalytics = useQuery(query, { variables, skip: !canQuery })
+  const boundaryAnalytics = useQuery<MapReportConstituencyStatsQuery>(query, {
```
Comment (ev-sc) [inline nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts:48]: I think it's too confusing and messy to leave it like this - to use a type belonging to a specific query (or at least named so).

File: nextjs/src/components/LocalisedMap.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ const LocalisedMap: React.FC<LocalisedMapProps> = ({
       mapStyle={
         showStreetDetails
           ? 'mapbox://styles/commonknowledge/clubx087l014y01mj1bv63yg8'
-          : 'mapbox://styles/commonknowledge/cm4cjnvff01mx01sdcmpbfuz5/draft'
+          : `mapbox://styles/commonknowledge/cm4cjnvff01mx01sdcmpbfuz5${process.env.NODE_ENV !== 'production' ? '/draft' : ''}`
```
Comment (ev-sc) [inline nextjs/src/components/LocalisedMap.tsx:38]: Nice 👍 I had forgotten about this

**Review by ev-sc — CHANGES_REQUESTED** [review]
The failing tests can be ignore, but there's just one thing that I think needs to be changed: using a type from a specific query for the return type of the `useBoundaryAnalaytics` hook.

File: nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts (hunk 2 above)
Comment (janbaykara) [reply nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts]: Check out the new changes, I'm quite happy with em :)

**Review by ev-sc — COMMENTED** [review]
@janbaykara I've just tried merging this into the branch I'm working on, and it's breaking something. I need to figure out what it is and come back to this review xx

**Comment by ev-sc** [comment]
@janbaykara Have you verified adding a new report and then trying to add data sources on your local machine? As opposed to just opening an existing report?
The latter works, but the former doesn't:


https://github.com/user-attachments/assets/a4d4d508-7ff7-41c0-8a6c-8e51294e281f

**Review by ev-sc — CHANGES_REQUESTED** [review]
Something is broken, but not clear what at the code level.
At the UI level: existing reports display correctly, but you can't add data sources to a new or existing report.

**Review by ev-sc — APPROVED** [review]
I approve

**Pull request merged**